packages/fxa-settings/src/lib/oauth/hooks.tsx

```diff
@@ -0,0 +1,229 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import AuthClient from 'fxa-auth-client/browser';
+import { useCallback } from 'react';
+import {
+  Integration,
+  OAuthIntegration,
+  isOAuthIntegration,
+} from '../../models';
+import { createEncryptedBundle } from '../crypto/scoped-keys';
+
+// TODO: should this live in lib/oauth/hooks or lib/hooks? Separate dir per hook?
```

Comment:
IMO, it we should group things by purpose first not by type first. We are following this generilazation elsewhere. e.g. we don't have a tests folder, rather we put the tests in the component folder near the component code. So in this case we put the oauth hooks in the oauth folder near the oauth code.

Reply:
Oh consistency.... such a tricky thing. You'll notice, there's a few comments on classes in the models folder about if they should go in the models folder or in the libs folder. For some reason this is making me question what I initially said above. 

We do in fact group by type in some regards. e.g. components, pages, models. Maybe hooks are their own top level type? They are very specific to react, but then again in this case the hook is also very specific to the oauth integration.  I guess I see arguments either way. I think most lines here are clearly oauth integration type code, so I guess I still stand by initial comment but am realizing that my opinion is a shaky here.

Reply:
Heh thanks for the follow up. We do have a ticket in the cleanup epic, FXA-7097, around moving hooks out from the index file and into their own directory in `hooks/`. I'll link this conversation there and we can address then.